framework/test/unit/state_machine/state_machine.spec.ts

```diff
@@ -205,6 +205,56 @@ describe('state_machine', () => {
 			expect(events).toHaveLength(1);
 			expect(events[0].toObject().topics[0]).toEqual(transaction.id);
 		});
+
+		it('should rollback state if afterCommandExecute fails', async () => {
+			const events = [
+				{
+					moduleID: utils.intToBuffer(3, 4),
+					typeID: Buffer.from([0, 0, 0, 0]),
+					data: utils.getRandomBytes(20),
+					topics: [utils.getRandomBytes(32), utils.getRandomBytes(20)],
+				},
+				{
+					moduleID: utils.intToBuffer(4, 4),
+					typeID: Buffer.from([0, 0, 0, 0]),
+					data: utils.getRandomBytes(20),
+					topics: [utils.getRandomBytes(32), utils.getRandomBytes(20)],
+				},
+				{
+					moduleID: utils.intToBuffer(2, 4),
+					typeID: Buffer.from([0, 0, 0, 0]),
+					data: utils.getRandomBytes(20),
+					topics: [utils.getRandomBytes(32)],
+				},
+			];
+			for (const e of events) {
+				eventQueue.add(e.moduleID, e.typeID, e.data, e.topics);
+			}
+
+			const snapshotID = eventQueue.createSnapshot();
+
+			const ctx = new TransactionContext({
+				eventQueue,
+				logger,
+				stateStore,
+				header,
+				assets,
+				networkIdentifier,
+				transaction,
+				currentValidators: [],
+				impliesMaxPrevote: true,
+				maxHeightCertified: 0,
+				certificateThreshold: BigInt(0),
+			});
+			await stateMachine.executeTransaction(ctx);
```

Comment:
I think to test if it rolls backs to the original test, we need to mock `before/afterCommandExecute` to insert some event, and fail on `afterCommandExecute`?

Reply:
Ahh yes, need to update the test.

Reply:
Updated!